### PR TITLE
use canonical_url specified in page if present

### PR DESCRIPTION
**Pull request by 8parth**
assign the canonical url to canonical_url specified by user in page otherwise use default generated url.

### CODE REVIEW

File: lib/jekyll-seo-tag/drop.rb [hunk 1]
```diff
@@ -183,7 +183,11 @@ def page_lang
 
       def canonical_url
         @canonical_url ||= begin
-          filters.absolute_url(page["url"]).to_s.gsub(%r!/index\.html$!, "/")
+          if page["canonical_url"].present?
```
Comment (benbalter) [inline lib/jekyll-seo-tag/drop.rb]: `.present?` only works in Rails. You'll want to use `to_s.empty?`

File: spec/jekyll_seo_tag/drop_spec.rb [hunk 2]
```diff
@@ -624,4 +624,23 @@
       end
     end
   end
+
+  context "canonical url" do
+    context "when canonical url is specified for a page" do
+      let(:page) { make_page( { "title" => "page title", "canonical_url" => "https://github.com/jekyll/jekyll-seo-tag/"} ) }
+      let(:canonical_url) { 'https://github.com/jekyll/jekyll-seo-tag/' }
+      it "uses specified canonical url" do
+        puts subject.canonical_url
```
Comment (benbalter) [inline spec/jekyll_seo_tag/drop_spec.rb]: Stray puts.

File: lib/jekyll-seo-tag/drop.rb (hunk 1 above)
Comment (8parth) [inline lib/jekyll-seo-tag/drop.rb]: Didn't know that. Will update this with specified changes.

File: spec/jekyll_seo_tag/drop_spec.rb (hunk 2 above)
Comment (8parth) [inline spec/jekyll_seo_tag/drop_spec.rb]: Forgot to remove puts, is this proper way to test or I need to create posts fixtures? I am not sure about this part :sweat_smile:

**Comment by benbalter** [comment]
@8parth this is great, and thanks for working through our feedback. I made a few minor formatting changes, but think this should be good to merge. Thank you!

**Pull request merged**